.github/workflows/release.yaml

```diff
@@ -287,6 +298,17 @@ jobs:
         GIT_COMMIT: ${{ github.sha }}
         GIT_TREE_STATE: clean
       run: make build-nightly-cli
+    - name: Sign and Notarize CLI
+      if: ${{ matrix.os == 'darwin' }}
+      env: 
+        QUILL_SIGN_P12: ${{ secrets.QUILL_SIGN_P12 }}
+        QUILL_SIGN_PASSWORD: ${{ secrets.QUILL_SIGN_PASSWORD }}
+        QUILL_NOTARY_KEY: ${{ secrets.QUILL_NOTARY_KEY }}
+        QUILL_NOTARY_KEY_ID: ${{ secrets.QUILL_NOTARY_KEY_ID }}
+        QUILL_NOTARY_ISSUER: ${{ secrets.QUILL_NOTARY_ISSUER }}
+        KARGO_BIN_PATH: bin/kargo-cli/${{ needs.publish-image.outputs.unstable-version }}/${{ matrix.os }}/${{ matrix.arch }}/kargo
+      run: |
+        make sign-and-notarize-cli
```

Comment:
Added to the nightly build for unstable versions as well. Wasn't sure if we wanted this but aired on the side of caution. Can be removed if undesired.